cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/BasicPullStrategyTest.java

```diff
@@ -379,4 +390,86 @@ public void pull_indexesUpdated()
 
     }
 
+    @Test
+    public void pull_changesNewerThanOpenRevs() throws Exception {
+        // test for regressions against the following scenario:
+        // - changes feed tells us that 1-rev is a new rev we don't have
+        // - on remote db, 2-rev gets added and becomes the new leaf
+        // - compact() is called and 1-rev is compacted away
+        // - we try to pull 1-rev but it is no longer there
+        // the solution to this is to re-fetch the changes feed at the same checkpoint and continue
+        // replication - in the hope that the changes feed now reflects the revs available
+
+        // upload some docs
+        Bar bar1 = BarUtils.createBar(remoteDb, "Tom", 31);
+
+        // capture 'real' changes feed - after first rev
+        final ChangesResult changesResult1 = remoteDb.changes(null, 1000);
+
+        // upload some more docs
+        Bar bar2 = BarUtils.updateBar(remoteDb, bar1.getId(), "Jerry", 41);
+
+        // capture 'real' changes feed - after second rev
+        final ChangesResult changesResult2 = remoteDb.changes(null, 1000);
+
+        // compact away the first rev
+        System.out.println("compact");
```

Comment:
I'm not sure what the sync tests use, would `java.util.Logger` be more appropriate?